lib/state/ui/reducer.ts

```diff
@@ -391,6 +391,15 @@ const showRevisions: A.Reducer<boolean> = (state = false, action) => {
   }
 };
 
+const showDeletedTags: A.Reducer<boolean> = (state = true, action) => {
+  switch (action.type) {
+    case 'DELETED_TAGS_TOGGLE':
+      return !state;
+    default:
+      return state;
+  }
+};
```

Comment:
if this is intended to be stored as an account-wide setting should we move it into the settings page? otherwise do we need this in state or would it be just as sufficient but simpler to keep the toggle state inside the component itself? do we expect that restoring deleted tags from one revision implies restoring tags from the next revision?

Reply:
also this action name confuses me. I think it means "toggle whether or not to restore deleted tags when restoring a note revision" but it sounds more like "show deleted tags"

`TOGGLE_RESTORING_DELETED_TAGS`??

Reply:
>if this is intended to be stored as an account-wide setting should we move it into the settings page?

Not sure, I'd say that it's not an account-wide settings, actually I'd expect that when I open the history screen, the "restore deleted tags" checkbox is always checked.

>otherwise do we need this in state or would it be just as sufficient but simpler to keep the toggle state inside the component itself?

This value is used for two things, one is for determining if the deleted tags should be restored when the user triggers the restore action and the other is to display the deleted tags of the current selected note's revision. Each one is handled in different components rendered in the `AppLayout` component, since they're not connected that's why I added it to the global state.

To be honest this won't be necessary and we could keep it in a component's state but it would imply to refactor the `RevisionSelector` component and add it as a child of `NoteRevisions` new component. I'll take a look into this and review the required changes.

>do we expect that restoring deleted tags from one revision implies restoring tags from the next revision?

I don't think so, I'll change this because I'd expect that when I open the history screen, the "restore deleted tags" checkbox is always checked, same as I commented regarding considering this an account-wide setting.

>also this action name confuses me. I think it means "toggle whether or not to restore deleted tags when restoring a note revision" but it sounds more like "show deleted tags"

It's used for both actually but I agree it can be confusing, I'll rename it to `TOGGLE_RESTORING_DELETED_TAGS` and update the action creator and reducer 👍 .

Reply:
>otherwise do we need this in state or would it be just as sufficient but simpler to keep the toggle state inside the component itself?

@dmsnell I created a new [PR](https://github.com/Automattic/simplenote-electron/pull/2844) that contains all the required changes to move the toggle state inside the `NoteRevisions` component. Let me know if you prefer to continue with that approach, thanks 🙇 !

Reply:
> one is for determining if the deleted tags should be restored when the user triggers the restore action and the other is to display the deleted tags of the current selected note's revision

that makes sense and I think is valid justification for it. I see you're already flipping it back to `true` when opening the history panel; a nice touch. honesty I don't think that we have to move everything into component state given your answer and given how much work is involved in  #2844 - I don't want to imply contorting ourselves in an inconvenient way to get this done - it was really a small question, "would this be easier in component state?" not realized it's used for both purposes.

Reply:
Yep I agree, I think we can continue with this option using Redux instead of moving to the component state approach (I'll close https://github.com/Automattic/simplenote-electron/pull/2844).

I decided to explore the component state approach as an exercise to see how easy it was to refactor that part. When I started to implement the `NoteRevisions` component, one thing that came to my mind was about encapsulating part of the revisions logic there, like the revisions selector. However I didn't want to change too much the structure as the original issue was to fix a bug 😄 .

Let me know if the current state of this PR is ok to finish the review, I'll fix the conflicting file in the meantime.
Thank you very much for all the work you did in reviewing this PR, I really do appreciate it  😊  .

Reply:
it's in a good state. please fix the conflict - I think we can get this in

Reply:
Ok, thanks! I've just fixed the conflict 🔧 .